app/controllers/region_admin/food_types_controller.rb

```diff
@@ -44,26 +37,63 @@ def create
 
   def edit
     @food_type = FoodType.find(params[:id])
+    @available_regions = available_regions
+
     authorize! :update, @food_type
-    @action = "update"
+
     session[:my_return_to] = request.referer
-    render :edit
   end
 
   def update
     @food_type = FoodType.find(params[:id])
+
     authorize! :update, @food_type
-    # can't set admin bits from CRUD controls
+
     if @food_type.update_attributes(params[:food_type])
       flash[:notice] = "Updated Successfully."
       unless session[:my_return_to].nil?
         redirect_to(session[:my_return_to])
       else
-        index
+        redirect_to region_admin_food_types_url
       end
     else
       flash[:error] = "Update failed :("
       render :edit
     end
   end
+
+  def destroy
+    @l = FoodType.find(params[:id])
+    authorize! :destroy, @l
+    @l.active = false
+    @l.save
+    redirect_to(request.referrer)
+  end
+
+  private
+
+  def available_food_types
+    if current_volunteer.super_admin?
+      FoodType.active
+    else
+      FoodType.active.regional(current_volunteer.region_ids)
+    end
+  end
+
+  def available_regions
+    if current_volunteer.super_admin?
+      Region.all
+    else
+      current_volunteer.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+    end
+  end
+
+  def authorize_region_admin!
+    return if region_admin?
+    redirect_to root_url, alert: "Unauthorized"
+  end
+
+  def region_admin?
+    current_volunteer.any_admin?
```

Comment:
How about changing `Ability#region_admin?` to public and delegating to that here to reduce duplicated logic?